File: examples/eApi v1.9/dotNET/CsobGTWDemo/CsobGatewayClientExample/Communication/DataObjects/Order.cs
using System.Text;
using Newtonsoft.Json;

namespace CsobGatewayClientExample.Communication.DataObjects;

public class Order : ApiBase
{
    [JsonProperty("type")] public string? Type { get; set; } //[purchase, balance, prepaid, cash, check 

    [JsonProperty("availability")] public string? Availability { get; set; }

    [JsonProperty("delivery")]
    public string? Delivery { get; set; } //[shipping, shipping_verified, instore, digital, ticket, order 

    [JsonProperty("deliveryMode")] public string? DeliveryMode { get; set; } //[0, 1, 2, 3 

    [JsonProperty("deliveryEmail")] public string? DeliveryEmail { get; set; }

    [JsonProperty("nameMatch")] public bool NameMatch { get; set; }

    [JsonProperty("addressMatch")] public bool AddressMatch { get; set; }

    [JsonProperty("billing")] public Address? Billing { get; set; }

    [JsonProperty("shipping")] public Address? Shipping { get; set; }

    [JsonProperty("shippingAddedAt")] public string? ShippingAddedAt { get; set; }

    [JsonProperty("reorder")] public bool Reorder { get; set; }

    [JsonProperty("giftcards")] public GiftCards? Giftcards { get; set; }


    public string? ToSign()
    {
        var sb = new StringBuilder();
        Add(sb, Type);
        Add(sb, Availability);
        Add(sb, Delivery);
        Add(sb, DeliveryMode);
        Add(sb, DeliveryEmail);
        Add(sb, NameMatch);
        Add(sb, AddressMatch);
        if (Billing != null) Add(sb, Billing.ToSign());
        if (Shipping != null) Add(sb, Shipping.ToSign());
        Add(sb, ShippingAddedAt);
        Add(sb, Reorder);
        if (Giftcards != null) Add(sb, Giftcards.ToSign());
        DeleteLast(sb);
        return sb.ToString();
    }
}

public class GiftCards : SignBase
{
    [JsonProperty("totalAmount")] public long? TotalAmount { get; set; }

    [JsonProperty("currency")] public string? Currency { get; set; }

    [JsonProperty("quantity")] public int? Quantity { get; set; }

    public override string ToSign()
    {
        var sb = new StringBuilder();
        Add(sb, TotalAmount);
        Add(sb, Currency);
        Add(sb, Quantity);
        DeleteLast(sb);
        return sb.ToString();
    }
}

File: examples/eApi v1.9/java/eAPI-v19-connector/src/main/java/cz/monetplus/mips/eapi/v19/connector/entity/Order.java
package cz.monetplus.mips.eapi.v19.connector.entity;

import com.fasterxml.jackson.annotation.JsonInclude;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

@Data
@EqualsAndHashCode(callSuper = false)
@JsonInclude(value = JsonInclude.Include.NON_NULL)
@AllArgsConstructor @NoArgsConstructor
class Order extends ApiBase implements Signable {
    private String type; //[purchase, balance, prepaid, cash, check]
    private String availability;
    private String delivery; //[shipping, shipping_verified, instore, digital, ticket, order]
    private String deliveryMode; //[0, 1, 2, 3]
    private String deliveryEmail;
    private boolean nameMatch;
    private boolean addressMatch;
    private Address billing;
    private Address shipping;
    private String shippingAddedAt;
    private boolean reorder;
    private GiftCards giftcards;

    @Override
    public String toSign() {
        StringBuilder sb = new StringBuilder();
        add(sb, type);
        add(sb, availability);
        add(sb, delivery);
        add(sb, deliveryMode);
        add(sb, deliveryEmail);
        add(sb, nameMatch);
        add(sb, addressMatch);
        if (billing != null) add(sb, billing.toSign());
        if (shipping != null) add(sb, shipping.toSign());
        add(sb, shippingAddedAt);
        add(sb, reorder);
        if (giftcards != null) add(sb, giftcards.toSign());
        deleteLast(sb);
        return sb.toString();
    }

    @Data
    @EqualsAndHashCode(callSuper = false)
    public static class GiftCards extends ApiBase implements Signable {
        private long totalAmount;
        private String currency;
        private int quantity;

        @Override
        public String toSign() {
            StringBuilder sb = new StringBuilder();
            add(sb, totalAmount);
            add(sb, currency);
            add(sb, quantity);
            deleteLast(sb);
            return sb.toString();
        }
    }
}
